src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -0,0 +1,765 @@
+                    break;
+                case TK_ANNOTATION:
+                    ret_code = get_annotation_dependencies(type_object.minimal().annotation_type(), type_dependencies);
+                    break;
+                case TK_STRUCTURE:
+                    ret_code = get_structure_dependencies(type_object.minimal().struct_type(), type_dependencies);
+                    break;
+                case TK_UNION:
+                    ret_code = get_union_dependencies(type_object.minimal().union_type(), type_dependencies);
+                    break;
+                case TK_SEQUENCE:
+                    ret_code = get_sequence_array_dependencies(type_object.minimal().sequence_type(),
+                                    type_dependencies);
+                    break;
+                case TK_ARRAY:
+                    ret_code = get_sequence_array_dependencies(type_object.minimal().array_type(), type_dependencies);
+                    break;
+                case TK_MAP:
+                    ret_code = get_map_dependencies(type_object.minimal().map_type(), type_dependencies);
+                    break;
+                // No dependencies
+                case TK_BITSET:
+                case TK_ENUM:
+                case TK_BITMASK:
+                    break;
+            }
+            break;
+        case EK_COMPLETE:
+            switch (type_object.complete()._d())
+            {
+                case TK_ALIAS:
+                    ret_code = get_alias_dependencies(type_object.complete().alias_type(), type_dependencies);
+                    break;
+                case TK_ANNOTATION:
+                    ret_code = get_annotation_dependencies(type_object.complete().annotation_type(), type_dependencies);
+                    break;
+                case TK_STRUCTURE:
+                    ret_code = get_structure_dependencies(type_object.complete().struct_type(), type_dependencies);
+                    break;
+                case TK_UNION:
+                    ret_code = get_union_dependencies(type_object.complete().union_type(), type_dependencies);
+                    break;
+                case TK_SEQUENCE:
+                    ret_code = get_sequence_array_dependencies(type_object.complete().sequence_type(),
+                                    type_dependencies);
+                    break;
+                case TK_ARRAY:
+                    ret_code = get_sequence_array_dependencies(type_object.complete().array_type(), type_dependencies);
+                    break;
+                case TK_MAP:
+                    ret_code = get_map_dependencies(type_object.minimal().map_type(), type_dependencies);
+                    break;
+                // No dependencies
+                case TK_BITSET:
+                case TK_ENUM:
+                case TK_BITMASK:
+                    break;
+            }
+            break;
+    }
+    return ret_code;
+}
+
+void TypeObjectRegistry::add_dependency(
```

Comment:
Being private API the chacks are not required if the developer uses the API as it is intended. The given TypeIdentifier should be valid because before calling this method, it should be checked that it is a direct hash TypeIdentifier (as it is done). If the TypeIdentifier is not known by the Registry, then someone has broken the contract, as the types registered into the Registry should always have previously registered their dependencies (the code generated by Fast DDS-Gen requires this) and if the TypeObject is built by the user, the API checks that the given types are known in Debug build mode.